lib/logstash/filters/mutate.rb

```diff
@@ -240,98 +248,91 @@ def rename(event)
       next unless event.include?(old)
       event[new] = event.remove(old)
     end
-  end # def rename
+  end
 
-  private
   def update(event)
     @update.each do |field, newvalue|
       next unless event.include?(field)
       event[field] = event.sprintf(newvalue)
     end
-  end # def update
+  end
 
-  private
   def replace(event)
     @replace.each do |field, newvalue|
       event[field] = event.sprintf(newvalue)
     end
-  end # def replace
+  end
 
   def convert(event)
     @convert.each do |field, type|
       next unless event.include?(field)
       original = event[field]
       # calls convert_{string,integer,float,boolean} depending on type requested.
-      converter = method("convert_" + type)
-      if original.nil?
-        next
-      elsif original.is_a?(Hash)
-        @logger.debug("I don't know how to type convert a hash, skipping",
-                      :field => field, :value => original)
-        next
-      elsif original.is_a?(Array)
-        value = original.map { |v| converter.call(v) }
+      converter = method(CONVERT_PREFIX + type)
+
+      case original
+      when Hash
+        @logger.debug? && @logger.debug("I don't know how to type convert a hash, skipping", :field => field, :value => original)
+      when Array
+        event[field] = original.map { |v| converter.call(v) }
+      when NilClass
+        # ignore
       else
-        value = converter.call(original)
+        event[field] = converter.call(original)
       end
-      event[field] = value
     end
-  end # def convert
+  end
 
   def convert_string(value)
     # since this is a filter and all inputs should be already UTF-8
     # we wont check valid_encoding? but just force UTF-8 for
     # the Fixnum#to_s case which always result in US-ASCII
     # see https://twitter.com/jordansissel/status/444613207143903232
-    return value.to_s.force_encoding(Encoding::UTF_8)
-  end # def convert_string
+    value.to_s.force_encoding(Encoding::UTF_8)
```

Comment:
perhaps add my jruby comment - in case someone in the future reckons that a check would be good/safer.